build/gen_stub.php

```diff
@@ -2979,6 +2980,30 @@ public function getVariableName(): string {
     }
 }
 
+/**
+ * @param array{0: string, 1: \PhpParser\Node\Arg[]}[] $attributes
+ * @param iterable<ConstInfo> $allConstInfos
+ */
+function generateAttributesCode(array $attributes, string $invocation, string $nameSuffix, iterable $allConstInfos): string {
+    $code = "";
+    foreach ($attributes as list($name, $args)) {
+        $escapedAttributeName = strtr($name, '\\', '_');
+        $code .= "\tzend_string *attribute_name_{$escapedAttributeName}_$nameSuffix = zend_string_init(\"" . addcslashes($name, "\\") . "\", sizeof(\"" . addcslashes($name, "\\") . "\") - 1, 1);\n";
+        $code .= "\t" . ($args ? "zend_attribute *attribute_{$escapedAttributeName}_$nameSuffix = " : "") . "$invocation, attribute_name_{$escapedAttributeName}_$nameSuffix, " . count($args) . ");\n";
+        $code .= "\tzend_string_release(attribute_name_{$escapedAttributeName}_$nameSuffix);\n";
+        foreach ($args as $i => $arg) {
+            $value = EvaluatedValue::createFromExpression($arg->value, null, null, $allConstInfos);
```

Comment:
I'm wondering if you ran the `./build/gen_stub.php` command, aren't you? I believe that it would be not longer possible to run gen_stub.php selectively for the stubs which register attributes (e.g. `./build/gen_stub.php ext/oci8`) because of the dependency to `Attribute::TARGET_METHOD`. I don't say that it's a huge problem for php-src itself, but the situation is not ideal. And I don't know if extensions will be able to register attributes this way because AFAIK `zend_attributes.stub.php` is not always available for them.... :(

Reply:
Yes, I was running that one. I think this is probably a general issue in the longer term, if there are any cross-dependencies.
Obviously if it were just for php-src one could add a require statement to note that dependency.
For extensions, the same could be possible, if we were to distribute our stub files with the headers... maybe?

Reply:
> Obviously if it were just for php-src one could add a require statement to note that dependency.

Yeah, I'll try implementing the support for `require` later as a follow-up.

> For extensions, the same could be possible, if we were to distribute our stub files with the headers... maybe?

I'm not sure how to do it, but it seems like a good idea :)